app/models/page/DashboardPage.scala

```diff
@@ -88,35 +139,121 @@ object DashboardPage {
     }
   }
 
-  def fromUser(user: CachedData)(implicit request: RequestHeader, lang: Lang): DashboardPage = status(user) match {
+  private def isApplicationSubmittedAndNotWithdrawn(user: CachedData)(implicit request: RequestHeader, lang: Lang) =
+    WithdrawApplicationRole.isAuthorized(user)
+
+  private def isApplicationInProgressAndNotWithdrawn(user: CachedData)(implicit request: RequestHeader, lang: Lang) =
+    CreatedOrInProgressRole.isAuthorized(user)
+
+  private def isApplicationWithdrawn(user: CachedData)(implicit request: RequestHeader, lang: Lang) =
+    WithdrawnApplicationRole.isAuthorized(user)
+
+  private def isApplicationCreatedOrInProgress(user: CachedData)(implicit request: RequestHeader, lang: Lang) =
+    PersonalDetailsRole.isAuthorized(user)
+
+  private def isUserWithNoApplication(user: CachedData)(implicit request: RequestHeader, lang: Lang) =
+    ApplicationStartRole.isAuthorized(user)
```

Comment:
I think the need to wrap these methods reflects an issue with the name of these 'Roles' - perhaps it's a step too far for this changeset but we should probably think about making "Personal Details Role" more descriptive i.e. "IsApplicationCreateOrInProgressRole" -- and probably change the word Role to something else.

Spotting the difference between "WithdrawRole" and "Withdraw*n*Role" posed a problem for us for a while